### ISSUE TITLE
Migrate the media & static settings to the new unified `STORAGES`

### ISSUE THREAD

**Issue by browniebroke** [cookiecutter/cookiecutter-django#4443]
## Description

Django 4.2 introduced a unified [`STORAGES` setting](https://docs.djangoproject.com/en/4.2/ref/settings/#std-setting-STORAGES) which unifies static & media setup and simplifies adding a custom storage that might be required by an application (e.g. private media).

## Rationale

The [`DEFAULT_FILE_STORAGE`](https://docs.djangoproject.com/en/4.2/ref/settings/#default-file-storage) and [`STATICFILES_STORAGE`](https://docs.djangoproject.com/en/4.2/ref/settings/#staticfiles-storage) settings are now both deprecated and we should migrate to the new setting before Django 5.1

**Comment by chipndell** [comment]
Hello @browniebroke ,
Currently I am working on it and it passed all the [Templates Test Suit](https://github.com/cookiecutter/cookiecutter-django/blob/f10253a3b3645d9b2629bc1cd8cecd1cefd1467f/CONTRIBUTING.md#run-the-templates-test-suite). But [Generate project tests](https://github.com/cookiecutter/cookiecutter-django/blob/f10253a3b3645d9b2629bc1cd8cecd1cefd1467f/CONTRIBUTING.md#run-the-generated-project-tests) fails with `django.db.utils.OperationalError: connection is bad: No such file or directory` error message. (using `test_bare.sh`)
Do we need any database running for these tests?

**Comment by browniebroke** [comment]
Yes, this test script needs a database. You can see how it's used in our CI workflow, there is service which is started.

It's great that you're trying to test locally, but don't spend too much time on the non-docker tests, the setup can be quite complicated, they'll run on GitHub when you open a pull request.

Thanks!
